src/include/buf_inline.h

```diff
@@ -88,6 +88,24 @@ __wt_buf_set(WT_SESSION_IMPL *session, WT_ITEM *buf, const void *data, size_t si
     return (__wt_buf_grow(session, buf, size));
 }
 
+/*
+ * __wt_buf_dup --
+ *     Duplicate the contents of one item into another
+ */
+static WT_INLINE int
+__wt_buf_dup(WT_SESSION_IMPL *session, WT_ITEM *dest, WT_ITEM *src)
+{
+    /* Clear the destination item structure it may have been created on the stack. */
+    memset(dest, 0, sizeof(WT_ITEM));
```

Comment:
Can this cause memory leak? If we have a bound previous set, we will lose the buffer. Should we just use __wt_buf_set instead?

Reply:
Yep - I started out having a safety check in here, but realized that I want to be able to pass an uninitialized WT_ITEM (hence the comment).

I'm not convinced it's better to push that uncertainty out to each caller that might want to copy the content of one WT_ITEM into a new WT_ITEM, but let me figure that out while I'm reviewing my change for memory safety..